engine/script/src/script.cpp

```diff
@@ -342,7 +342,12 @@ namespace dmScript
         {
             nwritten = dmSnPrintf(buffer, buffer_size, "  %s:%d: in function <%s:%d>\n", entry->short_src, entry->currentline, entry->short_src, entry->linedefined);
         }
-        return nwritten;
+
+        // Truncated, so we'll just make sure we've not written anything
+        if (nwritten < 0)
+            nwritten = 0;
```

Comment:
The fix.

Returning 0 means that the next entry will try to write on the same spot (and most likely fail)